compiler/noirc_driver/src/lib.rs

```diff
@@ -312,6 +318,7 @@ pub fn compile_main(
     crate_id: CrateId,
     options: &CompileOptions,
     cached_program: Option<CompiledProgram>,
+    emit_ssa: Option<PathBuf>,
```

Comment:
If we allow users to specify the path in the cli argument directly we won't need to pass a separate parameter here.

We could still default to the same file if it is not specified as well.

Reply:
I do also like restricting all artifacts to being generated within the `target` directory. Also makes management of errors and creation of directories simpler. We could also just operate under the assumption that the user has appropriately created the directory where they want to dump the SSA.

Reply:
> We could still default to the same file if it is not specified as well.

Do we want to emit the artifact even if it is not specified? I added the flag as to not have to always emit this artifact as it is only needed by very niche users.

Reply:
I ended up leaving this as is due to the reasons listed above. Although I understand about not wanting an extra parameter here, so I can change the flag to accept a user-provided directory if we feel strongly about that.

Reply:
> Do we want to emit the artifact even if it is not specified? I added the flag as to not have to always emit this artifact as it is only needed by very niche users.

I meant if a user specifies `--emit-ssa` instead of `--emit-ssa my/path/to/foo` we can default the path.

Reply:
> I still think we should refactor this a bit so that it is contained within CompileOptions and we don't need to manually pass around another flag.

Altering `CompileOptions` would require us to do some clones of that object if we are fine with that as substitute for this additional flag.

Reply:
I wouldn't think we'd need to clone CompileOptions if we can default the value right away

Reply:
I'm fetching the default path from the resolved workspace packages?

Reply:
I think we will have to as `CompileOptions` is per workspace while we want something that is per package.

Reply:
I'm going to try a different strategy for fetching the package build path.

Reply:
I'm going to look at including some of this workspace info in our compilation context as to avoid this extra flag.

Reply:
I ended up including a `package_build_path` field in our HIR `Context` object. This keeps us from having to thread a flag throughout `compile_main` but we now pass a `Workspace` to the `compile_program` nargo methods.